.github/workflows/integration_test_run.yaml

```diff
@@ -250,8 +253,9 @@ jobs:
           path: ${{ inputs.working-directory }}
           pattern: '*-charm'
           merge-multiple: true
+
       - name: Integration tests variable setting
-        working-directory: ${{ inputs.working-directory }}
+        working-directory: ${{ inputs.charm-working-directory || inputs.working-directory }}
```

Comment:
I was thinking that perhaps the `charm-working-directory` could be expressed as `charm-directory`, which would represent the relevant path to the charm directory from the working directory. The default value would be `charm-directory : "."`, and for charms located in `./charm/`, it would be `charm-directory: "charm"`. With this, the final charm working directory would be `{{ inputs.working-directory }}/{{ inputs.charm-directory }}`.

However, I believe the current solution is perfectly fine as well, so it's up to you.